src/interfaces/IERC721Permit.sol

```diff
@@ -18,10 +16,8 @@ interface IERC721Permit {
     /// @param spender The account that is being approved
     /// @param tokenId The ID of the token that is being approved for spending
     /// @param deadline The deadline timestamp by which the call must be mined for the approve to work
-    /// @param v Must produce valid secp256k1 signature from the holder along with `r` and `s`
-    /// @param r Must produce valid secp256k1 signature from the holder along with `v` and `s`
-    /// @param s Must produce valid secp256k1 signature from the holder along with `r` and `v`
-    function permit(address spender, uint256 tokenId, uint256 deadline, uint256 nonce, uint8 v, bytes32 r, bytes32 s)
+    /// @param signature Concatenated data from a valid secp256k1 signature from the holder, i.e. abi.encodePacked(r, s, v)
+    function permit(address spender, uint256 tokenId, uint256 deadline, uint256 nonce, bytes calldata signature)
```

Comment:
maybe worth an @dev to say why its payable lol

Reply:
sorry again kenny

Reply:
wait I added this in my other PR

Reply:
where u left a similar comment haah, so i got this one